/*
 * Copyright 2023 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package mmc;

import "mmc_config.proto";

option optimize_for = LITE_RUNTIME;

message CodecInitRequest {
  // Codec-specific parameters passed by client.
  ConfigParam config = 1;
}

message CodecInitResponse {
  // Socket name generated by the daemon.
  string socket_token = 1;
  // Input frame size accepted by the codec server.
  int32 input_frame_size = 2;
}
